modules/text/src/text_detector.cpp

```diff
@@ -0,0 +1,169 @@
+#include "precomp.hpp"
+#include "opencv2/imgproc.hpp"
+#include "opencv2/core.hpp"
+
+
+
+#include <iostream>
+#include <fstream>
+#include <sstream>
+#include <queue>
+#include <algorithm>
+#include <iosfwd>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+
+//#ifdef HAVE_CAFFE
+//#include "caffe/caffe.hpp"
+//#endif
+
+namespace cv { namespace text {
+
+
+
+
+class textDetectImpl: public textDetector{
+private:
+    struct NetOutput{
+        //Auxiliary structure that handles the logic of getting bounding box and confidences of textness from
+        //the raw outputs of caffe
+        Rect bbox;
+        float probability;
+
+
+        static void getOutputs(const float* buffer,int nbrTextBoxes,int nCol,std::vector<NetOutput>& res,Size inputShape)
+        {
+
+            res.resize(nbrTextBoxes);
+            for(int k=0;k<nbrTextBoxes;k++)
+            {
+                float x_min = buffer[k*nCol+3]*inputShape.width;
+                float y_min = buffer[k*nCol+4]*inputShape.height;
+                float x_max = buffer[k*nCol+5]*inputShape.width;
+                float y_max = buffer[k*nCol +6]*inputShape.height;
+                x_min = x_min<0?0:x_min;
+                y_min = y_min<0?0:y_min;
+                x_max = x_max> inputShape.width?inputShape.width-1:x_max;
+                y_max = y_max > inputShape.height?inputShape.height-1:y_max;
+                float wd = x_max-x_min+1;
+                float ht = y_max-y_min+1;
+
+                res[k].bbox=Rect(int(x_min),int(y_min),int(wd),int(ht));
+
+                res[k].probability=buffer[k*nCol+2];
+            }
+
+        }
+
+
+    };
+protected:
+
+    Ptr<TextRegionDetector> classifier_;
+public:
+    textDetectImpl(Ptr<TextRegionDetector> classifierPtr):classifier_(classifierPtr)
+    {
+
+    }
+
+
+
+    void textDetectInImage(InputArray inputImage,CV_OUT std::vector<Rect>& Bbox,CV_OUT std::vector<float>& confidence)
+    {
+                Mat netOutput;
```

Comment:
Need to follow the opencv c++ style guide for indentation.

Reply:
updated